great_expectations/rule_based_profiler/parameter_builder/simple_date_format_string_parameter_builder.py

```diff
@@ -0,0 +1,151 @@
+    ParameterBuilder,
+)
+from great_expectations.rule_based_profiler.types import (
+    Domain,
+    ParameterContainer,
+    build_parameter_container,
+)
+from great_expectations.validator.validator import Validator
+
+logger = logging.getLogger(__name__)
+
+
+class SimpleDateFormatStringParameterBuilder(ParameterBuilder):
+    """
+    Detects the domain date format from a set of candidate date format strings by computing the
+    column_values.match_strftime_format.unexpected_count metric for each candidate format and returning the format that
+    has the lowest unexpected_count ratio.
+    """
+
+    CANDIDATE_STRINGS: Set[str] = {
+        "%Y-%m-%d",
+        "%m-%d-%Y",
+        "%y-%m-%d",
+        "%Y-%m-%dT%z",
+        "%Y-%m-%d %H:%M:%S",
+    }
+
+    def __init__(
+        self,
+        name: str,
+        metric_domain_kwargs: Optional[Union[str, dict]] = None,
+        threshold: float = 1.0,
+        candidate_strings: Optional[Iterable[str]] = None,
+        additional_candidate_strings: Optional[Iterable[str]] = None,
+        data_context: Optional["DataContext"] = None,
+        batch_request: Optional[Union[BatchRequest, RuntimeBatchRequest, dict]] = None,
+    ):
+        """
+        Configure this SimpleDateFormatStringParameterBuilder
+        Args:
+            name: the name of this parameter -- this is user-specified parameter name (from configuration);
+            it is not the fully-qualified parameter name; a fully-qualified parameter name must start with "$parameter."
+            and may contain one or more subsequent parts (e.g., "$parameter.<my_param_from_config>.<metric_name>").
+            threshold: the ratio of values that must match a format string for it to be accepted
+            candidate_strings: a list of candidate date format strings that will REPLACE the default
+            additional_candidate_strings: a list of candidate date format strings that will SUPPLEMENT the default
+            data_context: DataContext
+            batch_request: specified in ParameterBuilder configuration to get Batch objects for parameter computation.
+        """
+        super().__init__(
+            name=name,
+            data_context=data_context,
+            batch_request=batch_request,
+        )
+
+        self._metric_domain_kwargs = metric_domain_kwargs
+
+        self._threshold = threshold
+        if candidate_strings is not None:
+            self._candidate_strings = set(candidate_strings)
+        else:
+            self._candidate_strings = self.CANDIDATE_STRINGS
+
+        if additional_candidate_strings is not None:
+            self._candidate_strings.update(additional_candidate_strings)
+
+    def _build_parameters(
+        self,
+        parameter_container: ParameterContainer,
+        domain: Domain,
+        *,
+        variables: Optional[ParameterContainer] = None,
+        parameters: Optional[Dict[str, ParameterContainer]] = None,
+    ):
```

Comment:
I just carried this forward from [metric_multi_batch_parameter_builder.py](https://github.com/great-expectations/great_expectations/blob/develop/great_expectations/rule_based_profiler/parameter_builder/metric_multi_batch_parameter_builder.py) and [numeric_metric_range_multi_batch_parameter_builder.py](https://github.com/great-expectations/great_expectations/blob/develop/great_expectations/rule_based_profiler/parameter_builder/numeric_metric_range_multi_batch_parameter_builder.py), so I would be interested to hear @alexsherstinsky's take on how this might be used and what type annotation is appropriate.